### PR TITLE
OWNERS(justaugustus): Prune extraneous reviewer roles

### PR DESCRIPTION
**Pull request by justaugustus**
Taking an opportunity to prune myself from some aliases/review paths
to reduce my workload.

If:
- I am no longer reviewing in an area, prune
- I was already an approver for the area, prune from review path

OWNERS_ALIASES are also updated to match their analogs in k/community.

Signed-off-by: Stephen Augustus <foo@auggie.dev>

### CODE REVIEW

**Comment by justaugustus** [comment]
/assign @BenTheElder @spiffxp @fejta 

FYI to various groups
RelEng:
@kubernetes/release-engineering 

Cloud Provider Azure:
@kubernetes/provider-azure 

Cluster API Azure:
@alexeldeib @CecileRobertMichon @devigned @nader-ziada

File: images/kubekins-e2e/OWNERS [hunk 1]
```diff
@@ -5,7 +5,6 @@ reviewers:
 - BenTheElder
 - cpanato
 - jeremyrickard
-- justaugustus
```
Comment (BenTheElder) [inline images/kubekins-e2e/OWNERS:8]: hmm we generally prefer to have approvers be a subset of reviewers, to spread the review load 🥕

Comment (dims) [reply images/kubekins-e2e/OWNERS:8]: +1

Comment (justaugustus) [reply images/kubekins-e2e/OWNERS]: Fixed.

Comment (justaugustus) [reply images/kubekins-e2e/OWNERS]: > hmm we generally prefer to have approvers be a subset of reviewers, to spread the review load 🥕

But... if we feel this way as a project, we should probably update https://go.k8s.io/owners#maintaining-owners-files:

> Good examples of OWNERS usage:
>
> - team aliases are used that correspond to sigs
> - there are more `reviewers` than `approvers`
> - the `approvers` are not in the `reviewers` section
> - OWNERS files that are regularly updated (at least once per release)

Comment (BenTheElder) [reply images/kubekins-e2e/OWNERS]: Thanks for fixing.

At the very least this repo / sig testing operates this way. I would encourage all repos to do so, unclear if the org as a whole will do this and I'm a bit weary of trying to effect org-wide policy at the moment.

File: config/jobs/kubernetes/generated/OWNERS [hunk 2]
```diff
@@ -2,7 +2,6 @@
 
 reviewers:
 - release-engineering-reviewers
-- justaugustus
 approvers:
 - release-engineering-approvers
 - justaugustus
```
Comment (BenTheElder) [inline config/jobs/kubernetes/generated/OWNERS:7]: same thing here? https://github.com/kubernetes/test-infra/pull/24328#discussion_r747707785

**Review by BenTheElder — COMMENTED** [review]
also feel free to make an entry in emeritus_approvers, which I believe is documented (opt-in) policy now?

File: config/jobs/kubernetes/generated/OWNERS (hunk 2 above)
Comment (justaugustus) [reply config/jobs/kubernetes/generated/OWNERS]: Fixed!

**Review by BenTheElder — APPROVED** [review]
thanks for cleaning up ALIASES and for your patience, and your reviews! :-)
/lgtm
/approve

**Pull request merged**